src/main/java/rx/schedulers/Schedulers.java

```diff
@@ -32,7 +35,14 @@
     private final Scheduler ioScheduler;
     private final Scheduler newThreadScheduler;
 
-    private static final Schedulers INSTANCE = new Schedulers();
+    private static Schedulers INSTANCE;
+
+    private static synchronized Schedulers getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Schedulers();
+        }
+        return INSTANCE;
+    }
```

Comment:
Alright. Do you think there's any risk of deadlock since we'd be calling it from `getInstance()` and it calls `getInstance()` internally?